NakedJavaScript/mensa-projekt
# M41 forgot password function

# What has changed?

I added phpmailer as a new mailing system for our App. With this it's very easy sending mails in various situations. I also created a test google mail account for the "company" which sends all the mail. I will also redesign the contact form later so that we'll also use phpmailer in there and not a 3rd party. 

@WeZyBugFix : 
Can you check the Alerts please? I tried to use them but it didn't work for me, so I just hardcoded them.

Also note: if the mail in the DB is written in uppercase like My_mail@mail.de but you type in my_mail@mail.de in the Form you might get an error or it doesn't work. So I thought the easiest and fastes way to solve that would just be formatting all mails in the DB to lowercase. I added that to the create-function of users @WeZyBugFix can you also add this to your edit function? 

Or if someone knows a better solution: feel free to implement it.

# How to test?

* Grab the new Database files
* Go to the index, you ofc need to be logged out
* Click on the "Passwort vergessen?" link
* In the new form type in your mail (a mail you've got access to), try a mail that exist and a mail that doesn't, in both cases you should receive a message (success and warning note)
* Once you've typed in a valid mail go to the database, you should now see that the "token" field now has a string in it
* Go to your mail program, check your mailbox. Note: the mail might end up in your spam folder, so check that as well
* You'll receive a mail with a link, open this link
* Check the link it should have your mail in it and the token
* type your new password
* Go to the Database, the token field should now be empty and you should have a new hashed password
* Try to reload the link, you should receive an error message that the Link isn't valid anymore
* Login with your new password
